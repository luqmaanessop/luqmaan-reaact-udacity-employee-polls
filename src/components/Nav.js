import { Link } from "react-router-dom";
import {connect} from "react-redux";
import {logoutAuthedUser} from '../actions/authedUser'

const Nav = ({dispatch, authedUserId}) => {

    const logout = (e) => {
        e.preventDefault();

        dispatch(logoutAuthedUser());
    };
    return (
        <nav className="flex items-center justify-between bg-blue-500 p-4">
            <div className="flex space-x-4">
                <Link to="/" className="text-white">Home</Link>
                <Link to="/leaderboard" className="text-white">Leaderboard</Link>
                <Link to="/new" className="text-white">New Poll</Link>
            </div>
            <span className="text-white" data-testid="user-information">User: {authedUserId}</span>
            <button onClick={logout} className="text-white">Logout</button>
        </nav>
    );
};


const mapStateToProps = ({authedUser}) => ({
  authedUserId: authedUser.id,
});


export default connect(mapStateToProps)(Nav);
