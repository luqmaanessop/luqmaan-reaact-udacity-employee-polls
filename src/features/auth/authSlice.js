// import {LOGOUT_AUTHED_USER, SET_AUTHED_USER} from "../../actions/authedUser";

// export default function authedUser(state = null, action) {
//   switch (action.type) {
//     case SET_AUTHED_USER:
//       return action.authedUser;
//     case LOGOUT_AUTHED_USER:
//       return null;
//     default:
//       return state;
//   }
// }

import { createSlice } from "@reduxjs/toolkit";

const authedUser = createSlice({
  name: 'authedUser',
  initialState: {},
  reducers: {
    SET_AUTHED_USER(state, action) {
      return { ...action.payload }
    },
    LOGOUT_AUTHED_USER(state, action) {
      return {}
    }
  }
})

export const { SET_AUTHED_USER, LOGOUT_AUTHED_USER } = authedUser.actions;

export default authedUser.reducer;
