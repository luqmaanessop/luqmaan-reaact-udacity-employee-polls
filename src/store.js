import { configureStore } from '@reduxjs/toolkit'

import logger from "./middleware/logger";
import authedUser from "./reducers/authedUser";
import questions from "./reducers/questions";
import users from "./reducers/users";


const store = configureStore({reducer: {
  authedUser,
  questions,
  users
}, middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger)});

export default store;
