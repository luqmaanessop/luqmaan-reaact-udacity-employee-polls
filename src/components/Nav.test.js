import * as React from "react";
import Nav from "../components/Nav";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../store";
import {SET_AUTHED_USER} from "../features/auth/authSlice";

describe("Nav", () => {
  beforeAll(async () => {
    store.dispatch(SET_AUTHED_USER({id: "sarahedo"}));
  });

  it("should match snapshot", async () => {

    const view = render(
      <Provider store={store}>
          <BrowserRouter>
              <Nav/>
          </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  })

  it("Should display username of logged-in user", () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav/>
        </BrowserRouter>
      </Provider>
    );

    const userSpanElement = screen.getByTestId("user-information");
    expect(userSpanElement.textContent).toBe("User: sarahedo");
  });

  it("should render all links correctly", async () => {
    render(
      <Provider store={store}>
          <BrowserRouter>
              <Nav/>
          </BrowserRouter>
      </Provider>
    );

    const homeLink = screen.getByText(/home/i);
    const newPollLink = screen.getByText(/new poll/i);
    const leaderboardLink = screen.getByText(/leaderboard/i);

    expect(homeLink).toBeInTheDocument();
    expect(newPollLink).toBeInTheDocument();
    expect(leaderboardLink).toBeInTheDocument();
  })
});
