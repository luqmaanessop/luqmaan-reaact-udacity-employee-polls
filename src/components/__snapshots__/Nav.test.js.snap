// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sticky top-0 flex items-center justify-between bg-blue-500 p-4"
      >
        <div
          class="flex items-center"
        >
          <img
            alt="Profile"
            class="mr-1 h-6 w-6 rounded-full overflow-hidden"
          />
          <span
            class="text-white"
            data-testid="user-information"
          >
            User: 
            sarahedo
          </span>
        </div>
        <div
          class="flex"
        >
          <a
            class="text-white mr-4 hover:font-medium"
            href="/"
          >
            Home
          </a>
          <a
            class="text-white mr-4 hover:font-medium"
            href="/leaderboard"
          >
            Leaderboard
          </a>
          <a
            class="text-white mr-4 hover:font-medium"
            href="/add"
          >
            New Poll
          </a>
        </div>
        <button
          class="text-white"
        >
          Logout
        </button>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sticky top-0 flex items-center justify-between bg-blue-500 p-4"
    >
      <div
        class="flex items-center"
      >
        <img
          alt="Profile"
          class="mr-1 h-6 w-6 rounded-full overflow-hidden"
        />
        <span
          class="text-white"
          data-testid="user-information"
        >
          User: 
          sarahedo
        </span>
      </div>
      <div
        class="flex"
      >
        <a
          class="text-white mr-4 hover:font-medium"
          href="/"
        >
          Home
        </a>
        <a
          class="text-white mr-4 hover:font-medium"
          href="/leaderboard"
        >
          Leaderboard
        </a>
        <a
          class="text-white mr-4 hover:font-medium"
          href="/add"
        >
          New Poll
        </a>
      </div>
      <button
        class="text-white"
      >
        Logout
      </button>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
